pymbolic/primitives.py

```diff
@@ -683,6 +734,124 @@ def __iter__(self):
         raise TypeError("expression types are not iterable")
 
 
+# {{{ dataclasses support
+
+def augment_expression_dataclass(cls: type[DataclassInstance]) -> type[Expression]:
```

Comment:
Is it worth adding something like `expr_dataclass` (similar to `tag_dataclass` in pytools) to squish the two decorators together?

Reply:
Yeah! I wasn't aware of `dataclass_transform` when this PR first started. (Maybe it didn't even exist? Unsure.) At any rate, I've made use of it now, and it's unquestionably nicer. `_augment_expression_dataclass` still exists but is now private.